packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -117,13 +126,24 @@ class FloatingActionButton extends StatefulWidget {
   /// and width of 40.0 logical pixels.
   final bool mini;
 
+  /// The margin to keep around the floating action button when creating a
+  /// notch for it.
+  ///
+  /// See also:
+  ///
+  ///   * [BottomAppBar], a material design elements that shows a notch for the
+  ///   floating action button.
+  final double notchMargin;
+
   @override
   _FloatingActionButtonState createState() => new _FloatingActionButtonState();
 }
 
 class _FloatingActionButtonState extends State<FloatingActionButton> {
   bool _highlight = false;
 
+  VoidCallback clearNotchMaker;
```

Comment:
either this should be private, or highlight should be public, but we shouldn't have both private and public members on this State.

Whichever you decide to do, do the same for _BottomAppBarState

Reply:
(i think convention is to make these private, but it doesn't much matter since they are private classes)

Reply:
done